frontend/src/features/projects/dispose/components/columns.test.tsx

```diff
@@ -0,0 +1,40 @@
+import { EditableClassificationCell } from './columns';
+import React from 'react';
+import renderer from 'react-test-renderer';
+import configureMockStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import * as reducerTypes from 'constants/reducerTypes';
+import { Provider } from 'react-redux';
+import { useFormikContext } from 'formik';
+
+jest.mock('formik');
+(useFormikContext as jest.Mock).mockReturnValue({});
+
+const mockStore = configureMockStore([thunk]);
+const mockCellInfo: any = {
+  row: {
+    id: '1',
+  },
+};
+
+const store = mockStore({
+  [reducerTypes.LOOKUP_CODE]: {
+    lookupCodes: [
+      { type: 'PropertyClassification', name: 'Core Operational', id: 0 },
+      { type: 'PropertyClassification', name: 'Core Strategic', id: 1 },
+      { type: 'PropertyClassification', name: 'Surplus Active', id: 2 },
+      { type: 'PropertyClassification', name: 'Surplus Encumbered', id: 3 },
+    ],
+  },
+});
+
+it('renders only surplus active and encumbered', () => {
+  const tree = renderer
+    .create(
+      <Provider store={store}>
+        <EditableClassificationCell {...mockCellInfo} />
+      </Provider>,
+    )
+    .toJSON();
+  expect(tree).toMatchSnapshot();
```

Comment:
This test just checks that the rendered component matches the last snapshot on file. If a dev inadvertently changes the logic of the component to display say 'Core Operational' and updates the snapshot this test will pass. I think the test name is a bit misleading as it only checks the `it('renders ok')`

We can leave as-is for now but I'd like to see a "real" test created that checks the business rule here: "only shows surplus active and encumbered". Ideally you would render the component with react-test-library and assert the number of "option" elements or something like that. Thoughts?

Reply:
Oh yes, that is a good call. I had that originally but started to overthink and thought it was redundant. But the way you worded it there shows me that it is not redundant at all and will catch errors on change! Will change the name of the test and add another before I merge!